src/main/java/com/stage/users/serviceImpl/UserServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.stage.users.serviceImpl;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.stage.users.exception.ResourceNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.config.ConfigDataResourceNotFoundException;
+import org.springframework.stereotype.Service;
+
+import com.stage.users.Iservice.UserIservice;
+import com.stage.users.Model.User;
+import com.stage.users.Repository.UserRepository;
+
+@Service
+public class UserServiceImpl implements UserIservice {
+
+	@Autowired
+	UserRepository userRepo;
+
+	@Override
+	public User addUser(User user) {
+		return userRepo.save(user);
+	}
+
+	@Override
+	public User retrieveUserById(long id) {
+		return userRepo.findById(id).get();
+	}
+
+	@Override
+	public List<User> retrieveAllUsers() {
+		return userRepo.findAll();
+	}
+
+	@Override
+	public void deleteUser(long id) {
+		userRepo.deleteById(id);
+	}
+
+	@Override
+	public User updateUser(User user, long id) {
```

Comment:
One improvement that could be made is to handle the case where the Optional<User> returned by userRepo.findById(id) is empty. Currently, the code only handles the case where the Optional<User> is present, and throws an exception otherwise. 
Instead, you could use the **orElseThrow** method to throw a **ResourceNotFoundException** if the user is not found: